circuit_knitting_toolbox/circuit_cutting/cutting_decomposition.py

```diff
@@ -38,36 +42,38 @@ def partition_circuit_qubits(
         partition_labels: A sequence containing a partition label for each qubit in the
             input circuit. Nonlocal gates belonging to more than one partition
             will be replaced with QPDGates.
+        inplace: Flag denoting whether to copy the input circuit before acting on it
 
     Returns:
         The output circuit with each nonlocal gate spanning two partitions replaced by a
         :class:`TwoQubitQPDGate`
 
     Raises:
-        ValueError: Length of partition_labels does not equal the number of qubits in the circuit
-        ValueError: Cannot decompose gate which acts on more than 2 qubits
+        ValueError: The length of partition_labels does not equal the number of qubits in the circuit.
+        ValueError: Input circuit contains unsupported gate.
     """
     if len(partition_labels) != len(circuit.qubits):
         raise ValueError(
             f"Length of partition_labels ({len(partition_labels)}) does not equal the number of qubits "
             f"in the input circuit ({len(circuit.qubits)})."
         )
 
-    new_qc = circuit.copy()
+    if not inplace:
+        circuit = circuit.copy()
 
     # Find 2-qubit gates spanning more than one partition and replace it with a QPDGate.
-    for i, instruction in enumerate(new_qc.data):
+    for i, instruction in enumerate(circuit.data):
         if instruction.operation.name == "barrier":
             continue
-        qubit_indices = [new_qc.find_bit(qubit).index for qubit in instruction.qubits]
+        qubit_indices = [circuit.find_bit(qubit).index for qubit in instruction.qubits]
         partitions_spanned = {partition_labels[idx] for idx in qubit_indices}
         # Ignore local gates and gates that span only one partition
         if (
             len(qubit_indices) <= 1
             or len(partitions_spanned) == 1
             or isinstance(instruction.operation, Barrier)
         ):
-            continue
+            continue  # pragma: no cover
```

Comment:
I commented on this line in the commit, but I don't see it now.  In any case ... I just realized that this function is not meaningfully changed in this PR.  In fact, I've been seeing coverage claim that this line is missing coverage locally, even though it reaches 100% coverage in CI.  I suppose you must be experiencing this, too?  It's odd, because I swear that this line _must_ be hit -- we test things like barriers, etc., right?